llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -7538,6 +7498,18 @@ LoopVectorizationPlanner::executePlan(
 
   ILV.printDebugTracesAtEnd();
 
+  // Adjust branch weight of the branch in the middle block.
```

Comment:
LVP::executePlan() is already quite long, with various step enumerated. Should additional steps be listed, and/or folded away somehow? Specifically, this weight adjustment was also part of completeLoopSkeleton().

Reply:
Yes, unfortunately it only fits here at the moment, assigned a number he